src/opcodes/execution/log.rs

```diff
@@ -167,11 +167,16 @@ impl<const N: usize, E: VmEncodingMode<N>> DecodedOpcode<N, E> {
                     is_service: is_first_message,
                 };
 
-                let refund = vm_state.refund_for_partial_query(
-                    vm_state.local_state.monotonic_cycle_counter,
-                    &partial_query,
-                );
-                let ergs_refund = refund.refund();
+                let ergs_refund = if variant == LogOpcode::StorageRead {
```

Comment:
Remove a branch here, add assert to 0 for transient, and hide any manipulation with oracles and witness tracer under "refund query function". Same below